src/net/udp.rs

```diff
@@ -246,7 +256,7 @@ impl UdpSocket {
     pub async fn sendmsg_zc<T: BoundedBuf, U: BoundedBuf>(
         &self,
         io_slices: Vec<T>,
-        socket_addr: SocketAddr,
+        socket_addr: Option<SocketAddr>,
```

Comment:
Please update the sendmsg_zc comment to say that None can be used when the socket has been connected and that it may be more efficient when sending multiple datagrams to the same address. I don't think you have to mention why it may be more efficient. I figure it's enough to trace through the code and see that a boxed value can be skipped for the uring operation and I haven't looked, but the kernel may also have already done its address parsing when the socket is connected so it could help on the kernel side too.

Reply:
Maybe create new struct UdpSocketConnected to be returned from UdpSocket::connect? It will have mostly same methods, but `sendmsg_zc` and `send_to` will be not taking `socket_addr` at all?

Reply:
Please add that as a developer's comment so the idea isn't lost but I would rather not try to enforce this with a compiler. The io_uring device should return an error in those cases.

Reply:
Compile time error is better than runtime. Isn't it better not to provide incorrect API at all, rather than documenting quirks and asking users to be careful?

Reply:
Added comment as  requested